pkg/render/egressgateway/egressgateway.go

```diff
@@ -365,13 +366,13 @@ func (c *component) egwRole() *rbacv1.Role {
 		},
 	}
 
-	if c.config.Installation.KubernetesProvider == operatorv1.ProviderOpenShift {
+	if c.config.OpenShift {
 		role.Rules = append(role.Rules,
 			rbacv1.PolicyRule{
 				APIGroups:     []string{"security.openshift.io"},
 				Resources:     []string{"securitycontextconstraints"},
 				Verbs:         []string{"use"},
-				ResourceNames: []string{"privileged"},
+				ResourceNames: []string{OpenShiftSCCName},
```

Comment:
@sridhartigera As discussed, this is reduced from `privileged` (changed in [PR #2724](https://github.com/tigera/operator/pull/2724/files#diff-ccd1538e4ede8bf20ce972b30d6d077dd6130c81508cd5df29198a4a885f34eaR371)), to our custom SCC.